zksync_sdk/zksync_provider/transaction.py

```diff
@@ -19,44 +27,47 @@ def __init__(self, provider, transaction_hash: str):
         self.provider = provider
         self.transaction_hash = transaction_hash
 
-    async def await_committed(self, attempts: int = 10, attempts_timeout: int = 100):
-        status = TransactionStatus.FAILED
+    async def await_committed(self, attempts: int = 10, attempts_timeout: int = 100) -> TransactionResult:
+        status = TransactionResult(TransactionStatus.FAILED,
```

Comment:
Maybe throw an exception in case there is no receipt for tx? The timeout does not mean that the tx has failed, but merely that there are some problems on a server-side, which makes processing transactions longer than usual. 

Also, let's put the default values for `attempts` as some kind of infinity? The users should handle timeouts only in case they do know what is going on

Reply:
Design was to allow user to handle this case by it's side. because I do not know the sync timeouts etc( example : wait for approved transaction) can be hours.

Reply:
Ok